tests/drivers/stepper/shell/src/main.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2024 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/drivers/stepper/stepper_fake.h>
+#include <zephyr/fff.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/shell/shell_dummy.h>
+#include <zephyr/ztest.h>
+
+#define FAKE_STEPPER_NAME DEVICE_DT_NAME(DT_NODELABEL(fake_stepper))
+
+/* Global variables */
+static const struct device *const fake_stepper_dev = DEVICE_DT_GET(DT_NODELABEL(fake_stepper));
+
+DEFINE_FFF_GLOBALS;
+
+ZTEST(stepper_shell, test_stepper_enable)
+{
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+	int err;
+
+	err = shell_execute_cmd(sh, "stepper enable " FAKE_STEPPER_NAME " on");
+	zassert_ok(err, "failed to execute shell command (err %d)", err);
+	zassert_equal(fake_stepper_enable_fake.call_count, 1, "enable function not called");
+	zassert_equal(fake_stepper_enable_fake.arg0_val, fake_stepper_dev, "wrong device pointer");
+	zassert_equal(fake_stepper_enable_fake.arg1_val, true, "wrong enable value");
+
+	RESET_FAKE(fake_stepper_enable);
```

Comment:
code duplication, write a common function.

Reply:
function are not really possible since fff creates a struct for each fake call. Used a macro instead :^)